tests/migration_test.go

```diff
@@ -692,6 +686,8 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 				// Start a ISCSI POD and service
 				By("Starting an iSCSI POD")
 				iscsiIP := tests.CreateISCSITargetPOD(cd.ContainerDiskEmpty)
+				tests.SkipIscsiPVCWithIPv6Test(iscsiIP)
```

Comment:
As we are going to drop the ipv6 only cluster this skip is redundant.
My initial thought was to prepare the test to be dual stack (work with both of the pod ips) but skip just the ipv6 one. But let's make it even easier, change the description of the relevant tests to mention it tests only the ipv4 flow.
For example here - 
`Context("with a shared ISCSI Filesystem PVC using the ipv4 address of the ISCSI pod...`

Reply:
We are on the same page, I tried to make those test work for both IP families but I there are issues when using IPv6 for ISCSI PVCs (#4150 ), so until we figure out how to fix them I added a skip

I will update the context description in following commit

Reply:
if you remove the skip totally, wont it fail on kind ipv6 until the lane is removed ?
(unless you plan to hold this one or remove the kind in the same PR which isnt good imo)

Reply:
We dont want to remove this skip until single stack IPv6 lane is dropped

Reply:
"As we are going to drop the ipv6 only cluster this skip is redundant."
so this, its better than master have (the skip there is less accurate)
so i am confused a bit

Reply:
This test is a ipv4 only test.  For me it doesn't make sense to skip it in case the podIp is ipv6 one. The test should always test the ipv4 address of the pod. If the pod doesn't have ipv4 adrees it should be skipped. But since we don't have such a cluster with no ipv4 address on the pod we can assume the pod has ipv4 address and fail otherwise.

I would use [ipAddressess.GetPodIpByFamily](https://github.com/kubevirt/kubevirt/blob/6a8d40ed2956bfd5b3d8e75a4925e91fd6866904/tests/libnet/ipaddress.go#L8) to get the ipv4 address of the pod (you can rebase on to of the mentioned unmerged PR).

Another option is to fail the test if the podIp is not an ipv4 one. I like this option less because we will have to change the test one a cluster with ipv6 as the primary ip will be introduced.
But since we currently support only clusters with ipv4 as the primary one, this option is valid.

Reply:
i would just keep the PR as its now, because it will skip if the ip under testing is ipv6, and it can happen miss only dual stack with reverse order that doesnt exists.
we already have tests that have the same problem as well, they would just fail in case we would have a reverse cluster, but i dont see it happen in the near future, and until it will , the needed PRs will be merged and it will be easy to fix what needed.

this way we wont block, and not create dependency
wdyt?

Reply:
It doesn't make sense to keep this skip here. The test is ipv4 only, if is confusing to skip the test if the address is ipv6. Why would the address be ipv6 on an ipv4 test. 
Just to illustrate the problem. Once the isci ipv6 bug will be fixed, can you just remove the skip? Would this ipv4 test run with ipv4 address.
As I already wrote, since the test is ipv4 one, you need to make sure you [use the ipv4 address](https://github.com/kubevirt/kubevirt/blob/6a8d40ed2956bfd5b3d8e75a4925e91fd6866904/tests/libnet/ipaddress.go#L8).

Reply:
I dont expect it to work by just removing the skip when the IPv6 issue will be fixed, we cant know what changes there would be once it is resolved (e.g: API change).

Since this PR scope is removing IsIPv6Cluster as prepration for removing ipv6 only lane and ISCSI PV wont work with IPv6 for now, we should not waste time in adapting or changing ISCSI PV tests to support dual-stack.

We can remove `SkipIscsiPVCWithIPv6Test` function and use `SkipOnKindInfra` instead 
Then we will end up with those tests working with IPv4 address only on dual-stack and single ipv4 stack cluster
with explicit title.
What you think?

Reply:
> I dont expect it to work by just removing the skip when the IPv6 issue will be fixed, we cant know what changes there would be once it is resolved (e.g: API change).

We do know the test is ipv4 only so skipping it when the ip is ipv6 because ipv6 has a bug doesn't make any sense.
This test should never be invoked for ipv6 since it is ipv4 test. It has nothing to do with the bug.

> 
> Since this PR scope is removing IsIPv6Cluster as preparation for removing ipv6 only lane and ISCSI PV wont work with IPv6 for now, we should not waste time in adapting or changing ISCSI PV tests to support dual-stack.
> 

The lane was already removed.
I didn't ask you to adapt the test to support dual stack. The opposite is correct. I asked this test to be ipv4 only.
You can find [here](https://github.com/kubevirt/kubevirt/pull/4156/files#diff-d642ef9101fdb0671d818d73a1c3f977R991) an example of how we get the ip address from a pod.

I will summarize again the options -
1. Using the PodIp
2. Using [libnet.GetPodIpByFamily(pod, k8sv1.IPv4Protocol) ](https://github.com/kubevirt/kubevirt/blob/6a8d40ed2956bfd5b3d8e75a4925e91fd6866904/tests/libnet/ipaddress.go#L8) to get the ipv4 address of the pod.

I prefer the second option since we know that in the future the main pod ip address may be the ipv6 one and it is pretty cheap  not to reside on the fact currently the ipv4 address is the primary one.
I would also like all our ipv4 only tests to be aligned an use the same pattern.

> We can remove `SkipIscsiPVCWithIPv6Test` function and use `SkipOnKindInfra` instead
> Then we will end up with those tests working with IPv4 address only on dual-stack and single ipv4 stack cluster
> with explicit title.
> What you think?

I think the is no reason to do it since the ipv6 kind lane is already gone.

Reply:
Gotcha